sdks/python/apache_beam/io/avroio.py

```diff
@@ -554,11 +554,11 @@ def avro_union_type_to_beam_type(union_type: List) -> schema_pb2.FieldType:
   """
   if len(union_type) == 2 and "null" in union_type:
     for avro_type in union_type:
-      if avro_type in AVRO_PRIMITIVES_TO_BEAM_PRIMITIVES:
-        return schema_pb2.FieldType(
-            atomic_type=AVRO_PRIMITIVES_TO_BEAM_PRIMITIVES[avro_type],
-            nullable=True)
-    return schemas.typing_to_runner_api(Any)
+      if avro_type != "null":
+        beam_type = avro_type_to_beam_type(avro_type)
+        if beam_type.WhichOneof("type_info") == "atomic_type":
+          return schema_pb2.FieldType(
+              atomic_type=beam_type.atomic_type, nullable=True)
```

Comment:
I like this approach, it is simpler than what I was suggesting. With that said, are we able to just do something like this?

```suggestion
        beam_type.nullable = True
        return beam_type
```

That way we're not just handling this for atomic types but for all types

Reply:
All done, also had to tweak a test but seems like it's all working as intended.